Homework2/Task1/Extensions/ExtensionClass.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Task1.Classes;
+static class ExtensionClass
+{
+ public delegate T Addition<T>(T firstElement, T secondElement);
+ public static GenericDiagonalMatrix<T> AddMatrixes<T>(this GenericDiagonalMatrix<T> matrixOne, GenericDiagonalMatrix<T> matrixTwo, Addition<T> addition)
+ {
+  bool isSizeEqual = matrixOne.Size == matrixTwo.Size;
+  T[] elementsOfNewMatrix;
+
+  void SetNewMatrixSize(GenericDiagonalMatrix<T> parMatrix)
+  {
+   elementsOfNewMatrix = new T[parMatrix.Size];
+  }
+
+  void AddElementsWithPadding(GenericDiagonalMatrix<T> smallerMatrix, GenericDiagonalMatrix<T> biggerMatrix)
+  {
+   for (int i = 0; i < smallerMatrix.Size; i++)
+   {
+    elementsOfNewMatrix[i] = addition(smallerMatrix[i, i], biggerMatrix[i, i]);
+   }
+   for (int i = smallerMatrix.Size; i < biggerMatrix.Size; i++)
+   {
+    elementsOfNewMatrix[i] = default(T);
```

Comment:
should be 
addition(  biggerMatrixElement, default)

Reply:
When we adding two matrixes, biggest diagonal should be kept, u can not replace it with zeroes.